sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -275,10 +285,19 @@ export class EventHubClient {
   }
 
   /**
-   * Creates an EventHubProducer that can be used to send events to the Event Hub for which this client
-   * was created.
+   * Creates an Event Hub producer responsible for transmitting `EventData` to the Event Hub.
+   * Depending on the `options` specified, the producer may be created to allow event data to
+   * be automatically routed to an available partition or specific to a partition.
+   *
+   * Allowing automatic routing of partitions is recommended when:
+   *  - The sending of events needs to be highly available.
+   *  - The event data should be evenly distributed among all available partitions.
+   *
+   * If no partition is specified, the following rules are used for automatically selecting one:
+   *  1. Distribute the events equally amongst all available partitions using a round-robin approach.
+   *  2. If a partition becomes unavailable, the Event Hubs service will automatically detect it and forward the message to another available partition.
```

Comment:
This last section on the rules used by the service when no partition is specified feels unnecessary in client docs. It makes sense in conceptual docs, but not here

Reply:
The 2 rules say similar things compared to the 2 bullet points before them i.e high availability and even distribution. Therefore, can be combined

```
Allowing automatic routing of partitions is recommended when:
    - The sending of events needs to be highly available.
    - The event data should be evenly distributed among all available partitions.

If no partition is specified, the following rules are used for automatically selecting one:
   1. Distribute the events equally amongst all available partitions using a round-robin approach.
   2. If a partition becomes unavailable, the Event Hubs service will automatically detect it and forward the message to another available partition.
```